src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -146,10 +146,16 @@ private async Task CreateBlocksAsync()
 
                     this.logger.LogInformation("Waiting {0} seconds until block can be mined.", waitingTimeInSeconds );
 
-                    if (waitingTimeInSeconds > 0)
+                    // This is a loop to account for different IDateTimeProvider implementations.
```

Comment:
@bokobza @noescape00 if you guys could give an opinion on this part that would be great.

Could be a bit cleaner but this is the smallest change that works.

Reply:
This change introduces copy-pasted logic. 
If such logic is required for integration tests maybe it's better to create miner for integration tests that would behave differently in a way that tests require?

Reply:
It does introduce copy-pasted logic, to support different DateTimeProviders. It would be possible to `do while` it or similar - I can remove the copy paste but this is just simpler.

 Otherwise why are we injecting an `IDateTimeProvider` and then making assumptions about its implementation??

>maybe it's better to create miner for integration tests that would behave differently in a way that tests require?

I'm not on board with this, there are too many places where we do this using inheritance for tests and it's inflexible, we end up testing different implementations of core classes, and it's hard to maintain.

Reply:
To avoid "copy-pasted logic" I adjusted the PoAMiner